### PR TITLE
surgery tool refactor

### PR DESCRIPTION
**Pull request by Fikou**
## About The Pull Request

surgery tools now work on tool defines
advanced surgery tools now work like i made power tools work in #46846
chaplain's chainsaw weapons can now do surgery slowly (the admin only inquisition chainsaw sword can do surgery FASTLY)
the mecha drill, chainsaw and mounted chainsaw and CLAMP can now do surgery with 100% accuracy to "promote fun mechanics" haha

## Why It's Good For The Game

makes advanced surgery tools less shitcode, less abusable, and makes some fun mechanics reward you with high surgery chances

## Changelog
:cl:
tweak: chainsaw kind of weapons and the mecha drill and the CLAMP can now be used with 100% accuracy for surgery
refactor: surgery tools now work on defines, and advanced surgery tools now switch the tool behaviours instead of creating new items
/:cl:

### CODE REVIEW

**Comment by PKPenguin321** [comment]
horrible commit names tbh

File: code/modules/surgery/mechanic_steps.dm [hunk 1]
```diff
@@ -3,7 +3,7 @@
 	name = "unscrew shell"
 	implements = list(
 		TOOL_SCREWDRIVER		= 100,
-		/obj/item/scalpel 		= 75, // med borgs could try to unskrew shell with scalpel
+		TOOL_SCALPEL 			= 75, // med borgs could try to unskrew shell with scalpel
```
Comment (lordpidey) [inline code/modules/surgery/mechanic_steps.dm]: unskrew?

File: code/modules/surgery/tools.dm [hunk 2]
```diff
@@ -279,78 +241,61 @@
...
-	sharpness = IS_SHARP_ACCURATE
-	light_color = LIGHT_COLOR_GREEN
-
-/obj/item/circular_saw/advanced/Initialize()
-	. = ..()
-	set_light(2)
+	playsound(get_turf(user), 'sound/machines/click.ogg', 50, TRUE)
+	if(tool_behaviour == TOOL_SCALPEL)
+		tool_behaviour = TOOL_SAW
+		to_chat(user, "<span class='notice'>You incease the power of [src], now it can cut bones.</span>")
```
Comment (Arturlang) [inline code/modules/surgery/tools.dm]: Typo, increase, not incease, perhaps rewritting the word might help with legibility as well.

**Comment by zxaber** [comment]
>the mecha drill can now be used with 100% accuracy for surgery

Acquire Ripley. Become Mecha Dentist.

**Comment by nemvar** [comment]
I vaguely adding some shitty gun_saw global list. Maybe convert that too?

File: code/modules/surgery/tools.dm (hunk 2 above)
Comment (Fikou) [reply code/modules/surgery/tools.dm]: ertgasdha

File: code/modules/surgery/mechanic_steps.dm (hunk 1 above)
Comment (Fikou) [reply code/modules/surgery/mechanic_steps.dm]: in a comment and not my fault

**Pull request merged**